apps/core-app/src/components/customFields/RagequitTokenList.tsx

```diff
@@ -0,0 +1,212 @@
+import { useMemo } from 'react';
+import { useFormContext } from 'react-hook-form';
+import {
+  getNetwork,
+  formatValueTo,
+  memberTokenBalanceShare,
+  memberUsdValueShare,
+  NETWORK_TOKEN_ETH_ADDRESS,
+} from '@daohaus/common-utilities';
+import {
+  Buildable,
+  Field,
+  ParSm,
+  WrappedCheckbox,
+  Checkbox,
+  DataSm,
+} from '@daohaus/ui';
+
+import { useConnectedMembership, useDao } from '@daohaus/dao-context';
+import { CheckboxProps, CheckedState } from '@radix-ui/react-checkbox';
+import styled from 'styled-components';
+import { TokenBalance } from '@daohaus/dao-data';
+import { useParams } from 'react-router-dom';
+import { sortTokensForRageQuit } from '../../utils/general';
+
+const TokenListContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const Column = styled.div`
+  width: 33%;
+`;
+
+const DataColumn = styled(Column)`
+  display: flex;
+  flex-direction: column;
+  gap: 1.2rem;
+  padding-top: 0.8rem;
+`;
+
+type TokenTable = {
+  tokenCheckboxes: CheckboxProps[];
+  amounts: React.ReactNode[];
+  usdValue: React.ReactNode[];
+};
+
+export const RagequitTokenList = (props: Buildable<Field>) => {
+  const { id } = props;
+  const { dao } = useDao();
+  const { connectedMembership } = useConnectedMembership();
+  const { daochain } = useParams();
+  const { setValue, watch } = useFormContext();
+
+  const [sharesToBurn, lootToBurn, tokens] = watch([
+    'sharesToBurn',
+    'lootToBurn',
+    'tokens',
+  ]);
+
+  const networkData = useMemo(() => {
+    if (!daochain) return null;
+    return getNetwork(daochain);
+  }, [daochain]);
+
+  const handleSelectAll = (checked: CheckedState) => {
+    if (checked) {
+      setValue(
+        id,
+        sortTokensForRageQuit(
+          dao?.tokenBalances
+            .filter((token) => Number(token.balance) > 0)
+            .map((token) => token.tokenAddress || NETWORK_TOKEN_ETH_ADDRESS) ||
+            []
+        )
+      );
+    } else {
+      setValue(id, []);
+    }
+
+    dao?.tokenBalances.forEach((token) => {
+      if (Number(token.balance) > 0) {
+        setValue(token.tokenAddress || NETWORK_TOKEN_ETH_ADDRESS, checked);
+      }
+    });
+  };
+
+  const tokenTable = useMemo((): TokenTable | null => {
+    if (!dao || !networkData || !connectedMembership) return null;
+    return dao?.tokenBalances
+      .filter((token) => Number(token.balance) > 0)
+      .reduce(
+        (acc: TokenTable, token: TokenBalance) => {
+          acc.tokenCheckboxes = [
+            ...acc.tokenCheckboxes,
```

Comment:
im going to have to a get a breakdown of how you use reduce -- i usually pick other methods but this is really nice